src/app/components/daily-review/daily-review.component.ts

```diff
@@ -1,50 +1,328 @@
+
+      this.accountService.getCurrentDateWellnessEntry(
+        this.currentDate,
+        this.currentUserId
+      ).
+        pipe(take(1)).
+        subscribe(guidedJournalEntry => {
+
+          if (guidedJournalEntry.length !== 0) {
+            //update the existing currentDate wellness entry with the Forms data
+            this.journalRecordId = guidedJournalEntry[0].id;
+            guidedJournalFormData.id = this.journalRecordId;
+            this.accountService.updateGuidedjournalData(guidedJournalFormData).
+              pipe(take(1)).
+              subscribe(() => {
+                console.log("jounral form update success");
+                this.alertService.success('Journal Entry Successfully Updated');
+              });
+
+          }
+        });//end subscribe
+    }
+
+  }
+  submitHabitReviewForm() {
+    let habitLogs: IHabitCompletionLog[] = [];
+  
+    if (this.newHabitLogEntry === true) {
+      
+      Object.keys(this.habitReviewForm.controls).forEach(key => {
+        console.log(
+          "keys =",
+          key,
+          this.habitReviewForm.controls[key].value
+        );
+        habitLogs.push({
+          id: 0,
+          userId: this.currentUserId,
+          habitId: Number(key),
+          completed: this.habitReviewForm.controls[key].value,
+          date: this.currentDate
+        });
+      });
+      
+      this.accountService.addHabitCompletionLogs(habitLogs);
+
+    } else if (this.newHabitLogEntry === false) {
+     
+      console.log("form before controls = ", this.currentDateHabitLogs);
+      Object.keys(this.habitReviewForm.controls).forEach(key => {
+        this.currentDateHabitLogs.forEach((log) => {
+          if (key === log.habitId.toString()) {
+            log.completed = this.habitReviewForm.controls[key].value;
+          }
+        });
+        console.log("keys =", key, this.habitReviewForm.controls[key].value);
+
+      });
+    }
+    this.accountService.updateHabitCompletionLogs(this.currentDateHabitLogs);
+
+  }
+
+  submitWellnessRatingForm() {
+    const wellnessFormData: IWellnessRating = this.wellnessRatingForm.getRawValue();
+
+    if (this.newWellnessEntry === true) {
+
+      this.newWellnessEntry = false;
+
+      this.accountService.addWellnessRatingEntry(wellnessFormData)
+      .pipe(take(1))
+      .subscribe();
+    } else if (this.newWellnessEntry === false) {
+      console.log("new entry false");
+
+      this.accountService.getCurrentDateWellnessEntry(
+        this.currentDate,
+        this.currentUserId
+      ).
+        pipe(take(1)).
+        subscribe(wellnessEntry => {
+
+          if (wellnessEntry.length !== 0) {
+            //update the existing currentDate wellness entry with the Forms data
+            this.wellnessRecordId = wellnessEntry[0].id;
+            wellnessFormData.id = this.wellnessRecordId;
+            this.accountService.updateWellnessData(wellnessFormData).
+              pipe(take(1)).
+              subscribe(() => {
```

Comment:
same here. Can't subscribe within a subscribe. Is this common code that could be moved out to somewhere shared and reused?